intel-types/src/lib.rs

```diff
@@ -52,12 +54,267 @@ impl<T: Copy + Debug + PartialEq + BitAnd<Output = T>> PartialEq<T> for Masked<T
+    pub fdp: u64,
+
+    /// 32-bit register providing status and control bits used in SIMD floating-point operations
+    pub mxcsr: Masked<MxCsr>,
+
+    /// Register used to perform operations on 64-bit packed integer data
+    pub mm: [Mm; 8],
+
+    /// 128-bit data register used to operate on packed single-precision floating-point operands
+    pub xmm: [Xmm; 16],
+
+    /// Padding: Size of XSaveLegacy must be 512.
+    pub padding0: [u64; 11],
+
+    /// Padding: Size of XSaveLegacy must be 512.
+    pub padding1: [u8; 7],
+}
+
+impl Default for XSaveLegacy {
+    /// The x87 state initial configuration for FCW masks all floating-point exceptions,
+    /// sets rounding to nearest, and sets the x87 FPU precision to 64 bits.
+    /// See Section 8.1.5.
+    ///
+    /// The initial state of MXCSR after power-up/reset or INIT; mask allows software
+    /// to identify any reserved bits in MXCSR (none are reserved here).
+    /// See Sections 11.6.4 and 11.6.6.
+    fn default() -> Self {
+        let mut xsl = MaybeUninit::<XSaveLegacy>::zeroed();
+        unsafe {
+            (*xsl.as_mut_ptr()).fcw = Fcw::from_bits(0x037F).unwrap();
+        }
+        unsafe {
+            (*xsl.as_mut_ptr()).mxcsr = Masked::<MxCsr> {
+                data: MxCsr::from_bits(0x1F80).unwrap(),
+                mask: MxCsr::from_bits(0xFFF).unwrap(),
+            }
+        }
+        let xsl_init = unsafe { xsl.assume_init() };
+        xsl_init
+    }
+}
+
+/// The XSAVE header of an XSAVE area comprises the 64 bytes starting at offset 512 from the
+/// area's base address. See Section 13.4.2.
+#[repr(C)]
+pub struct XSaveHeader {
+    /// State-component bitmap identifying the state components in the XSAVE area.
+    pub xstate_bv: u64,
+
+    /// State-component bitmap indicating the format of the XSAVE extended region and whether
+    /// a component is in the XSAVE area.
+    pub xcomp_bv: XcompBv,
+
+    /// Reserved
+    pub reserved0: [u64; 6],
+}
+
+impl Default for XSaveHeader {
+    fn default() -> Self {
+        Self {
+            xstate_bv: Default::default(),
+
+            /// XCOMP_BV[63] = 1, compaction mode for the XSave Extended Region. However, no state
+            /// components are included in the Extended Region as all other bits are 0. The size of
+            /// the XSaveXtd is therefore zero.
+            /// TODO: Check this value
+            xcomp_bv: XcompBv::from_bits(0x80000000).unwrap(),
+
+            reserved0: Default::default(),
+        }
+    }
+}
+
+/// For our use case, the XSave Extended Region is in compacted format and currently holds nothing,
+/// as specifed in XCOMP_BV in
+/// https://github.com/jsakkine-intel/linux-sgx/blob/master/tools/testing/selftests/x86/sgx/encl_bootstrap.S#L89.
+/// For more on this region, see Section 13.4.3.
+#[derive(Debug, Default)]
+#[repr(C)]
+pub struct XSaveExtend;
+
+/// For details on the fields included in XSave, see Section 13.4.
+#[repr(C)]
```

Comment:
Is there an alignment requirement?

Reply:
"Each of the save and restore instructions takes a memory operand that specifies the 64-byte aligned base address of the XSAVE area on which it operates." Since the base address is 64-byte aligned, this means the entire XSAVE area is 64-byte aligned - is this the correct way to interpret this statement?